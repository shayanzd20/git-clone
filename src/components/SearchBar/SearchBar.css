.ui.search .input {
	width: 100%;
	margin-top: 20px;
}
.ui.search .prompt {
	border-radius: 5px;
	width: 100%;
	padding: 7px;
}
.ui.input > input {
	border: 1.5px solid rgba(34, 36, 38, 0.15);
}
i.icon {
	float: right;
	font-size: 20px;
	cursor: pointer;
}
