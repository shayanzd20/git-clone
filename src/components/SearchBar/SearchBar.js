import React, { useRef } from 'react';

import './SearchBar.css';

/**
 *
 * @description Search Bar Component
 * @param {string} term - term to search
 * @param {function} searchKeyword - function to filter repository
 * @returns {component}
 */
const SearchBar = ({ term, searchKeyword }) => {
	const inputEl = useRef('');

	const getSearchTerm = () => {
		searchKeyword(inputEl.current.value);
	};

	return (
		<div data-testid="search" className="ui search">
			<div className="ui icon input">
				<input id="data-inputid" ref={inputEl} type="text" placeholder="Search Repositories" className="prompt" value={term} onChange={getSearchTerm}></input>
				<i className="search icon-search"></i>
			</div>
		</div>
	);
};

export default React.memo(SearchBar);
