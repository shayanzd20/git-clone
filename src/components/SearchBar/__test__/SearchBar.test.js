import React from 'react';
import ReactDOM from 'react-dom';
import SearchBar from '../SearchBar';
import { configure, shallow, mount, simulate } from 'enzyme';
import { render, cleanup, act } from '@testing-library/react';
import Adapter from 'enzyme-adapter-react-16';

import renderer from 'react-test-renderer';
configure({ adapter: new Adapter() });

afterEach(cleanup);

const initialProps = {
	term: "'type anything here'",
};

describe('Search Bar', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = mount(<SearchBar />);
	});

	it('render without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<SearchBar />, div);
	});

	it('check data-testid', () => {
		expect(wrapper.find('div#data-testid')).toBeTruthy();
	});

	const { term } = initialProps;
	it('check place holder', () => {
		expect(wrapper.find(<SearchBar term={term} />)).toBeTruthy();
	});

	it('matches snapshot', () => {
		const tree = renderer.create(<SearchBar term={term} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
