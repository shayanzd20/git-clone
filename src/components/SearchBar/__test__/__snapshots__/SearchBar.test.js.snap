// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar matches snapshot 1`] = `
<div
  className="ui search"
  data-testid="search"
>
  <div
    className="ui icon input"
  >
    <input
      className="prompt"
      id="data-inputid"
      onChange={[Function]}
      placeholder="Search Repositories"
      type="text"
      value="'type anything here'"
    />
    <i
      className="search icon-search"
    />
  </div>
</div>
`;
